### DISCUSSION TITLE
AVR EB product Brief dropped

### DISCUSSION THREAD

**Discussion by MX682X** [SpenceKonde/DxCore#375] [General]
https://www.avrfreaks.net/s/topic/a5C3l000000UkyzEAC/t193040

https://onlinedocs.microchip.com/oxy/GUID-7AB01992-0E92-4D58-9CA9-DD27613C944B-en-US-4/GUID-A849C6C2-EF64-4024-8724-403C5D364C92.html

![grafik](https://user-images.githubusercontent.com/58419867/206922130-571dd7a6-b0fd-4fef-94d5-2983eecae534.png)

This sounds interesting, considering there is also a PLL (in the block diagram)...

**Comment by SpenceKonde** [comment]
Did someone swithch the poland springs water the break room with vodka over at Microchip? Or maybe LSD? Or did someone switch my bottled water for LSD and I'm hallucinating this whole thing? (Lousy trip if so) - just when it looked like they'd dedicated themselves to some measure of consistency they discard the two main existing types of timer in favor of two other wacky ones? And bringing back WEX? are they forgetting the lessons of xmega already?  Gaaaaa!

**Reply by MX682X** [reply]
I have no Idea what happend with WEX, but I guess better to have it then not? Also, it looks like TCE is just two TCAs with the same counter. And TCF just a TCB, but with less signal input logic?

Anyway, when I was looking how to drive a BLDC with an AVR Dx, I would have had to use some CCL logic to extract another channel out of the TCA. Now they can be used more easily by a direct connection. But I do think it's stupid that this thing has only 2 ACs. with three (and an external DAC I guess) they could have made an excelent current limit on all three phases.... (Basically the successor of Timer/Counter 4 of the old AVRs)

Basically, this thing is a motor driver with an CPU, in my opinion.

**Reply by SpenceKonde** [reply]
Well, they need a motor driver + CPU since the previous ones are all getting ancient. 

Where are you getting that information on what a TCE and TCF is? How did you get to that product brief? I'm wondering if there is a product brief for the DU-series available through the same weird corner of microchip's website. 

The info from the product brief falsifies what you claimed though - the TCE is capable of outputting 8 channels. A timer A can output either 3 or 6. So TCE is not two TCAs, though I suspect it has considerable similarity. If the TCE+WEX has the capabilities of a TCA generally, the huge question in everyone's mind would naturally be "8 channels. Does that mean in some sort of split mode analogy, or do you get the full 16 bits with each channel?
And for TCF, you forgot to mention that it's also 24 bits. Another case where it makes a BFD whether the PLL can be used to clock it, and what actual modes it has relating to input capture. 
It matters a great deal whether that PLL is actually there and which timers can use it. 

Agree re: ACs, though I don't see why you would need an external dac if you had 3 ACs (though you do with two). 
Ya wanna know what's really stupid? They dropped the fucking second USART! You gotta have 2 USARTs on a uC, man, otherwise you can't even set up a string of them using serial to communicate. That's important, I mean, what will you do when you have 1000 of them controlling th BLDC motors spinning the gas centrifuges enriching uranium for your clandestine nuclear weapons project, and to make it immune to attacks on the centralized control (like StuxNet - it's direct target was to find SCADA systems that looked like gas centrifuge facilities, make the BLDC motors misbehave so severely that they would spin themselves to pieces (filling the centrifuge hall with corrosive, toxic uranium hexaflouride), you want to decentralize it, so the centrifuges can tell the adjacent ones about conditions on their side so they can all optimize their operational parameters, making a self regulating enrichment cascade. But you couldn't do that with a single USART, because they'd need to talk to both the one ahead of them and behind them in the cascade....

2 USARTs is one of the highest priority features for me, honestly.

**Reply by MX682X** [reply]
Just guesses. 
I mean it's always easier to start with something than from scratch. And The EA already had two TCAs with two compare channels each, so it makes sense that one TCE with four channels is likely a combined Timer. (I also had in mind - three compares are enough, why making four, it makes less sense)
Same story with TCB - EA had 4 of them, minus two that are kept, leaves 2x16-bit TCBs of silicon space for the TCF. And since the TCB is already used as timer for millis (aka 1kHz frequency generation) I came to the conclusion that it's the most straight-forward thing they could do.

I found about it though AVR freaks and thought it would make sense to share it to another corner of the internet

**Reply by SpenceKonde** [reply]
Straightforward yes - except judging by the names, they're still digging through the xmega junkyard and picking off pieces of xmega, so it might be like what xmega has, modified to fit modern AVRs.

**Comment by MX682X** [comment]
![grafik](https://user-images.githubusercontent.com/58419867/218142282-91056330-2e3d-4524-b0c5-e69fa4f2a287.png)
AVR8, huh? 
8 Pin AVRs with 20MHz clock sounds suspiciously like a tiny

(Source: https://onlinedocs.microchip.com/?find=EB)

**Reply by SpenceKonde** [reply]
14 and 20 pin AVRs with 20 MHz clocks sound exactly like a tiny don't they?

**Reply by SpenceKonde** [reply]
Yum! Does this mean the DU is actually coming? 


avr16ea28, avr16ea32, avr16ea48, avr32ea28, avr32ea32, avr32ea48, avr64ea28, avr64ea32, avr64ea48, avr128da28, avr128da32, avr128da48, avr128da64, avr128db28, avr128db32, avr128db48, avr128db64, avr16dd14, avr16dd20, avr16dd28, avr16dd32, **avr16du14, avr16du20, avr16du28, avr16du32**, avr32da28, avr32da32, avr32da48, avr32db28, avr32db32, avr32db48, avr32dd14, avr32dd20, avr32dd28, avr32dd32, **avr32du14, avr32du32,** avr64da28, avr64da32, avr64da48, avr64da64, avr64db28, avr64db32, avr64db48, avr64db64, avr64dd14, avr64dd20, avr64dd28, avr64dd32, **avr64du28, avr64du32**

**Reply by MX682X** [reply]
> 14 and 20 pin AVRs with 20 MHz clocks sound exactly like a tiny don't they?
they do, but apparently there ain't enough market for 8-pin chips



> Yum! Does this mean the DU is actually coming?

I noticed that too. Seems so. And yet still no official product brief. 
I've got to admit I kinda wanna write the USB driver for that... Or maybe just optimize it. 
I like challenges, as I learn more the harder they are.... 
Can't be much harder then writing an AVR BTStack or a custom optimization pass for avr-gcc....

**Reply by SpenceKonde** [reply]
> And yet still no official product brief.

Euhm, that's not entirely true. Why don't you send me an email (spencekonde@gmail - I don't have your email address)

**Comment by SpenceKonde** [comment]
Other than that this is clearly positioned to eat tintavrs lunch confirming
my theory that tinyavr was always a stogap..., nor sure what to say

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Reply by MX682X** [reply]
I messed up. the first number was the flash size. whoops. Smallest pin number is 14 afterall.

But hey, maybe this will push the tinies down. I've got hopes for a remake of sot232-6 tinies like tiny 4/5/9/10 with UPDI and 3.3V Programming. Just that it probably never will come true... (they are perfect for making smart comparators and psuh-button logics)

**Reply by SpenceKonde** [reply]
I'd love it too... but they'd need to be AVRrc I think to fit in the small packages and have shit for memory. Did you notice what's missing from the DD line? There is no 64k 20-pin DD in VQFN, only Wide $OIC

I can concive of only one reason they'd make their largest flash member of a family jump off a cliff wrt relevance in the increasingly relevant sectors where QFNs are the package of choice.... And that is that the 64k die was too big. I should glue a 32DD20 to a surface and start filing away the top and see how much of the area inside the chip is occupied by the die... 

In fact, I think that's what I will do

**Reply by SpenceKonde** [reply]
If I can find a single container of superglue that hasn't hardened in the tube!!

Edit: Why the hell am I messing with glue, just solder one down to something ffs...

**Reply by MX682X** [reply]
To be fair, they would have to kinda double the die space to go from 32k to 64k. In fact Flash and SRAM are the two things using up like 80% of the die (and I/O about another 15%). 
And when it comes to sot-323-6 chips, there is only so much you can do with 3 GPIO and a Reset. You simply don't need that much memory.

Also, Is there a reason why noone melts the package, with a hot air gun for example?

**Reply by SpenceKonde** [reply]
Q. E. D!!

![QED](https://user-images.githubusercontent.com/6779483/218542708-ef235145-54bf-4943-92d0-a3729ae5e8bb.jpg)

That's a 32k die. I don't see a way to cram in twice the flash memory in that package, do you? 

What I'm saying is *Microchip is die size limited at the small end*. I bet this is why there was no qfn 3216 either. But apparently the 2-series either got a process shrink - or maybe that's why they took a chainsaw to the list of peripherals.

**Reply by SpenceKonde** [reply]
Also, that was a beautiful solder job to waste on this. No stencil, just a hand dispensed blob of good old tin lead solder by KEK (they are one of the better chinese brands, both in leaded and unleaded (SAC305). Soldered with hot air and hotplate, and a quick drag with a good soldering iron tip and it was perfect. Okay, yeah this was the 0,4% silver leaded, but still... 

I don't want to think about trying to do that with some crapola solder like SnBiAg 65:35:0.3-1. (pro tip: lead free solder advertised as melting at 183C either isn't, or doesn't). There is a solder that melts within 10 C of SnPb, though - and it's not awful. But I bet I still couldn't have done that quite so effortlessly. Maybe with one of Relife's 158C ones. They have two - one says "lead solder paste" but tests negative for lead, the other one has an ROHS logo but gives ambiguous results on lead tests), and the low melting point makes everything so much easier

**Reply by MX682X** [reply]
Good job. Ever considered getting a digital microscope?

**Reply by SpenceKonde** [reply]
Those sound expensive. 

So no I hadn't considered it. Would need to wrassle up some more moolah I think...

**Reply by SpenceKonde** [reply]
Hmm, well maybe there's more to it that this, because I had a 64k chip that had a short on the board somewhere, did a crap job of removing it, then soldered it to a crap pcb, and filed the top off... and the die was *smaller*. So either it was actually a 16k one instead of a 64k one (the soldermask scrawl on the back SAID 64k though!)

**Comment by SpenceKonde** [comment]
Check the capabilities of AVRxt vs AVRrc. I personally think they're doing away with ATtiny as a brand. Or they're going to make only RC tinies from here on out, and "tinyAVR" will come to mean reduced core ones and [capital-letter]x-series will be the full AVR core.
